cmd/goquery_completion/conditional.go

```diff
@@ -31,6 +31,86 @@ func openParens(tokens []string) int {
 	return open
 }
 
+// dirKeywordCount returns the number of dir keyword occurrences in tokens
+func dirKeywordCount(tokens []string) int {
+	dirKeywordCount := 0
+	for _, token := range tokens {
+		if token == types.FilterKeywordDirection || token == types.FilterKeywordDirectionSugared {
+			dirKeywordCount++
+		}
+	}
+	return dirKeywordCount
+}
+
+// firstUnnestedBinaryOp returns the position of the first 'unnested' binary logical
+// operator inside tokens and the respective token.
+// It returns -1 and an empty string if tokens do not contain any binary logical operator.
+func firstUnnestedBinaryOp(tokens []string) (int, string) {
```

Comment:
Here I'd keep it because your fallback return value isn't the null element of at least one of the parameters...